### ISSUE TITLE
[Bug]: Error: Couldn't reach supabase. Either the address doesn't exist or the network might not be on. Retrying in 10s

### ISSUE THREAD

**Issue by k0shk0sh** [supabase-community/supabase-kt#228]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/wiki/Troubleshooting) page for similar problems
- [X] I enabled [logging](https://github.com/supabase-community/supabase-kt/wiki/Logging) and checked the logs

### Version(s)

1.1.0

### Target(s)

JVM

### What happened? (include your code)

While returning to my project to work on, I re-run the app and I always get logged-out with the log shared below. 

Here is my setup: 

```
bindSingleton<SupabaseClient> {
        val (url, key) = SupabaseConfigModel()
        createSupabaseClient(
            supabaseUrl = url,
            supabaseKey = key
        ) {
            defaultSerializer = KotlinXSerializer(instance())
            install(GoTrue)
            install(Postgrest)
            install(Realtime)
            install(GraphQL) {
                apolloConfiguration {
                    addHttpInterceptor(LoggingInterceptor(LoggingInterceptor.Level.BODY))
                }
            }
            install(Functions)
            install(Storage)
        }
    }
 bindSingleton<GoTrue> { instance<SupabaseClient>().gotrue }
```

After few relaunches, the app works just fine and the error is gone, otherwise logging in fixes it for the meantime. 

### Steps To Reproduce (optional)

1. signIn:
```
override suspend fun signIn(email: String) {
        goTrue.sendOtpTo(Email, createUser = true) {
            this.email = email
        }
    }
```

2. verifyOtp:  
```
 goTrue.verifyEmailOtp(OtpType.Email.MAGIC_LINK, email, otp)
```

3. successful login. 
4. keep the desktop app open for awhile 
5. rerun the app fron Intellij IDE again. 
6. observe the behaviour of logging out the user and getting the error log below. 

### Relevant log output (optional)

```shell
Error: Couldn't reach supabase. Either the address doesn't exist or the network might not be on. Retrying in 10s
java.lang.NullPointerException: Cannot invoke "io.github.jan.supabase.network.KtorSupabaseHttpClient.request(String, kotlin.jvm.functions.Function1, kotlin.coroutines.Continuation)" because the return value of "io.github.jan.supabase.SupabaseClient.getHttpClient()" is null
	at io.github.jan.supabase.network.SupabaseApi.rawRequest$suspendImpl(SupabaseApi.kt:24)
	at io.github.jan.supabase.network.SupabaseApi.rawRequest(SupabaseApi.kt)
	at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApi.rawRequest(AuthenticatedSupabaseApi.kt:23)
	at io.github.jan.supabase.network.SupabaseApi.request(SupabaseApi.kt:20)
	at io.github.jan.supabase.gotrue.GoTrueImpl.refreshSession(GoTrueImpl.kt:641)
	at io.github.jan.supabase.gotrue.GoTrueImpl.importSession(GoTrueImpl.kt:382)
	at io.github.jan.supabase.gotrue.GoTrueImpl.loadFromStorage(GoTrueImpl.kt:429)
	at io.github.jan.supabase.gotrue.GoTrue.loadFromStorage$default(GoTrue.kt:301)
	at io.github.jan.supabase.gotrue.GoTrueImpl$1.invokeSuspend(GoTrueImpl.kt:79)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
```
```

**Comment by jan-tennert** [comment]
Publishing version `1.1.1-dev`, can you check if that exception still occurs?

**Comment by k0shk0sh** [comment]
Left the app open since yesterday when posting this, I did update to 1.1.1-dev and relaunched the app and i dont see the issue. I will keep monitoring and let you know if I encountered it again. 

Can you point me to the commit? im curious to see what was it 🙈

**Comment by jan-tennert** [comment]
https://github.com/supabase-community/supabase-kt/commit/2f20c6d8ae454634211b649f99d2f5aed2ed88bc. The only reason I could think of was that the plugins get created before the HTTP Client which cause this NPE

**Comment by k0shk0sh** [comment]
Hi Jan, 

now, the silent error is as below: 

```
Error: Couldn't reach supabase. Either the address doesn't exist or the network might not be on. Retrying in 10s
java.lang.NullPointerException: Cannot invoke "io.github.jan.supabase.plugins.PluginManager.getInstalledPlugins()" because "this_$iv" is null
	at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApi.rawRequest$lambda$1(AuthenticatedSupabaseApi.kt:70)
	at io.github.jan.supabase.network.KtorSupabaseHttpClient.request(KtorSupabaseHttpClient.kt:45)
	at io.github.jan.supabase.network.SupabaseApi.rawRequest$suspendImpl(SupabaseApi.kt:24)
	at io.github.jan.supabase.network.SupabaseApi.rawRequest(SupabaseApi.kt)
	at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApi.rawRequest(AuthenticatedSupabaseApi.kt:23)
	at io.github.jan.supabase.network.SupabaseApi.request(SupabaseApi.kt:20)
	at io.github.jan.supabase.gotrue.GoTrueImpl.refreshSession(GoTrueImpl.kt:641)
	at io.github.jan.supabase.gotrue.GoTrueImpl.importSession(GoTrueImpl.kt:382)
	at io.github.jan.supabase.gotrue.GoTrueImpl.loadFromStorage(GoTrueImpl.kt:429)
	at io.github.jan.supabase.gotrue.GoTrue.loadFromStorage$default(GoTrue.kt:301)
	at io.github.jan.supabase.gotrue.GoTrueImpl$1.invokeSuspend(GoTrueImpl.kt:79)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
```

**Comment by jan-tennert** [comment]
Made some more internal changes, can you try out version `1.1.1-dev-2`?

**Comment by k0shk0sh** [comment]
I'll do that later today and get back to you.

Thanks for the quick response and support 🙏

**Comment by k0shk0sh** [comment]
so far, I didn't encounter the issue, I will keep this open until Sunday when I will work again on the project im currently doing and keep you posted. 

Danke Jan.

**Comment by jan-tennert** [comment]
> so far, I didn't encounter the issue, I will keep this open until Sunday when I will work again on the project im currently doing and keep you posted.
> 
> Danke Jan.

Alright, included this fix in `1.1.1` for now!

**Comment by k0shk0sh** [comment]
so far, I haven't encounter the issue and also the auth issue seems to be gone.

Cheers.
